### PR TITLE
BREAKING CHANGE: Rename *HeapExo* to *Exo*

### PR DESCRIPTION
**Pull request by erights**
https://github.com/endojs/endo/pull/1459 has not yet been merged into the endo monorepo, so a global rename is still not as painful as it is about to be. Now that the heap exo level will be about to stand alone in `@ando/exo`, with the virtual and durable variants to remain (at least for now) in agoric-sdk, I am increasingly unhappy with the `Heap` adjective added to those names.

So I just added a commit to the endo PR, currently at https://github.com/endojs/endo/pull/1459/commits/7fc2d06830e81eb926d0c9386cc529b023f1a192 , which does the rename. In that PR were to be merged with that state, the heap-level exo names would be simpler.

This PR does the same rename over agoric-sdk, so things will work now, and things will continue to work when agoric-sdk is upgraded to depend on https://github.com/endojs/endo/pull/1459, agoric-sdk imports these from `@endo/exo`, and the local copy of exo is deleted.

### CODE REVIEW

**Comment by erights** [comment]
Reviewers, I recommend github's "Hide whitespace" especially for this PR.

File: packages/store/docs/exo-taxonomy.md [hunk 1]
```diff
@@ -17,22 +17,30 @@ We often call a record of named entangled Xs an "XKit", as it a "toolkit" is a c
...
 
 * ***`prepareExo`*** <br>
-Like `makeHeapExo` but for a durable exo in baggage.
+Like `makeDurableExo` but for a durable exo in baggage.
 * ***`prepareExoClass`*** <br>
-Like `defineHeapExoClass` but for a durable exo in baggage.
+Like `defineDurableExoClass` but for a durable exo in baggage.
 * ***`prepareExoClassKit`*** <br>
-Like `defineHeapExoClassKit` but for a durable exo in baggage.
+Like `defineDurableExoClassKit` but for a durable exo in baggage.
```
Comment (michaelfig) [inline packages/store/docs/exo-taxonomy.md]: I find this diff confusing if their original formulation was not a mistake.  Are all these sentences true?

**Review by michaelfig — APPROVED** [review]
LGTM!  Just one question about the docs.

File: packages/store/docs/exo-taxonomy.md (hunk 1 above)
Comment (erights) [reply packages/store/docs/exo-taxonomy.md]: Neither is untrue, but `Durable` was and is the better explanation in context. Resolving.

Comment (Chris-Hibbert) [reply packages/store/docs/exo-taxonomy.md]: I  echo @michaelfig's sentiment. How could any of these statements: "like ...Durable..., except that it's durable" be correct?

**Review by Chris-Hibbert — APPROVED** [review]
modulo @michaelfig's question on the doc, this LGTM.

File: packages/store/docs/exo-taxonomy.md (hunk 1 above)
Comment (erights) [reply packages/store/docs/exo-taxonomy.md]: Gotcha. Will rephrase.

Comment (erights) [reply packages/store/docs/exo-taxonomy.md]: Done. I reverted back to fixing the original phrasing. That's actually better than what I had in mind.

**Pull request merged**